R/zchunk_L241.trade_input.R

```diff
@@ -0,0 +1,194 @@
+    return(c("L241.StubAgTradeCoeff_food",
+             "L241.StubAgTradeCoeff_nonfood",
+             "L241.StubAgTradeCoeff_feed",
+             "L241.AgProdTech_RES_output",
+             "L241.RES_Market"))
+  } else if(command == driver.MAKE) {
+
+    all_data <- list(...)[[1]]
+
+    scenario <- year <- GCAM_region_ID <- region <- value <- GCAM_commodity <-
+      technology <- supplysector <- res.secondary.output <- market <-
+      output.ratio <- NULL # silence package check notes
+
+    # Load required inputs
+    GCAM_region_names <- get_data(all_data, "common/GCAM_region_names")
+    A_demand_technology <- get_data(all_data, "aglu/A_demand_technology")
+    A_an_input_technology <- get_data(all_data, "aglu/A_an_input_technology")
+    L101.ag_kcalg_R_C_Y <- get_data(all_data, "L101.ag_kcalg_R_C_Y")
+    L2012.AgProduction_ag_irr_mgmt <- get_data(all_data, "temp-data-inject/L2012.AgProduction_ag_irr_mgmt")
+    L102.pcgdp_thous90USD_Scen_R_Y <- get_data(all_data, "L102.pcgdp_thous90USD_Scen_R_Y")
+
+    # Build tables (lines 38-43 in old file)
+    # First, determine which regions are in which groupings
+    L102.pcgdp_thous90USD_Scen_R_Y %>%
+      filter(scenario == "SSP4", year == 2010) %>%
+      select(GCAM_region_ID, value) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      mutate(value = value * gdp_deflator(2010, base_year = 1990)) ->
+      L241.pcgdp_2010
+
+    L241.high_reg <- L241.pcgdp_2010$region[L241.pcgdp_2010$value > aglu.HIGH_GROWTH_PCGDP]
+    L241.low_reg <- L241.pcgdp_2010$region[L241.pcgdp_2010$value < aglu.LOW_GROWTH_PCGDP]
+
+    # Create table of regions, technologies, and all base years (47-51)
+    # NOTE: Easiest if the model base years are subsetted from a full table as a last step in the construction of each of these tables
+    A_demand_technology %>%
+      write_to_all_regions(c(aglu.NAMES_TECH, "minicam.energy.input", "market.name"), GCAM_region_names) %>%
+      mutate(stub.technology = technology) ->
+      A_demand_technology_R
+    A_an_input_technology %>%
+      write_to_all_regions(c(aglu.NAMES_TECH, "minicam.energy.input", "market.name"), GCAM_region_names) %>%
+      mutate(stub.technology = technology) ->
+      A_an_input_technology_R
+
+    # Add lookup vectors to level1 output tables (55-59)
+    # Add region names to Level1 data tables
+    aglu_demand_calyears <- HISTORICAL_YEARS[HISTORICAL_YEARS %in% MODEL_YEARS]
+    aglu_demand_futureyears <- MODEL_YEARS[!MODEL_YEARS %in% aglu_demand_calyears]
+    L101.ag_kcalg_R_C_Y %>%
+      filter(year %in% aglu_demand_calyears) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      filter(year == max(aglu_demand_calyears)) %>%
+      select(region, GCAM_commodity, value) ->
+      L241.ag_kcalg_R_C_Yf.mlt
+
+    # Fraction of food that should be produced locally
+    LOCAL_FOOD_FRACT <- 0.8
+
+    # Coefficient for ag_trade of food crops (incl secondary products) (61-69)
```

Comment:
I think this needs some additional comments. Essentially, what we are doing is requiring that some fraction (80%) of all food is produced domestically. We use the `RES` policy code in GCAM to do this. The two chunks below set up the demand portion of that...requiring the demand sectors to demand `ag_trade` credits in addition to crops/livestocks. The demand is equal to 0.8 * the amount of crop/livestock demanded to ensure that the 80% threshold is met.